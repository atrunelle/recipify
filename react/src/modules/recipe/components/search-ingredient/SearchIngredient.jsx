import './SearchIngredient.css';

import React, { Component } from 'react';
import { Typography, TextField, Button, Card, CardContent, Grid, CircularProgress } from '@material-ui/core';

class SearchIngredient extends Component {
  constructor(props) {
    super(props);

    this.state = {
      ingredientName: '',
      numberOfServings: 1,
    };
  }

  onIngredientChange = (event) => {
    this.setState({ ingredientName : event.target.value });
  }

  onNumberOfServingsChange = (event) => {
    this.setState({ numberOfServings : event.target.value });
  }

  onSubmit = (event) => {
    event.preventDefault();
    this.props.add(this.state.ingredientName, this.state.numberOfServings);
  }
  
  render() {
    let buttonContent = '';

    if (this.props.isFetching) {
      buttonContent = <CircularProgress style={{ color: '#fff' }} />;
    } else {
      buttonContent = 'Add ingredient';
    }

    return (
      <Card className="u-push-top">
        <CardContent>
          <Typography
            gutterBottom 
            variant="headline"
            component="h2">
            What do you have in your fridge?
          </Typography>          
          <form className="search" onSubmit={this.onSubmit}>
            <Grid 
              container
              spacing={8}
              justify="space-between">
              <Grid
                item
                xs={12} 
                sm={6}>
                <TextField
                  fullWidth
                  label="Find an ingredient"
                  value={this.state.ingredientName} 
                  id="ingredientNameInput"
                  placeholder="Ex: 200gr chicken"
                  onChange={this.onIngredientChange}/>
              </Grid>
              <Grid item>
                <TextField
                  label="Number of servings"
                  value={this.state.numberOfServings}
                  id="numberOfServingsInput"
                  onChange={this.onNumberOfServingsChange}/>
              </Grid>
              <Button
                disabled={!this.state.ingredientName}
                type="submit"
                variant="raised"
                color="primary"
                id="submitButton">
                {buttonContent}
              </Button>
            </Grid>
          </form>
        </CardContent>
      </Card>
    );
  }
}

export default SearchIngredient;
