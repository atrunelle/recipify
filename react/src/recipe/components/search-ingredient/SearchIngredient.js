import './SearchIngredient.css';

import React, { Component } from 'react';

class SearchIngredient extends Component {
    constructor(props) {
        super(props);
    
        this.state = {
          ingredientName: '',
          numberOfServings: 1,
        };
    }

    onIngredientChange = (event) => {
      this.setState({ ingredientName : event.target.value });
    }

    onNumberOfServingsChange = (event) => {
      this.setState({ numberOfServings : event.target.value });
    }

    onSubmit = (event) => {
      event.preventDefault();
      this.props.add(this.state.ingredientName, this.state.numberOfServings);
    }
    
  render() {
    return (
        <form className="search" onSubmit={this.onSubmit}>
          <label className="search__label">
            What do you want to use?
          </label>
          <input 
            className="search__input" 
            value={this.state.ingredientName} 
            type="text" 
            placeholder="Find an ingredient..."
            onChange={this.onIngredientChange}/>
          <input 
            className="search__input" 
            value={this.state.numberOfServings} 
            type="text" 
            placeholder="Find an ingredient..."
            onChange={this.onNumberOfServingsChange}/>
            <button type="submit">Submit</button>
        </form>
    );
  }
}

export default SearchIngredient;
