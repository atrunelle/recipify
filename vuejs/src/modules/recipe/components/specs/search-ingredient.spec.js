import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import SearchIngredient from '../search-ingredient';
import apiService from '@/core/api.service';

describe('Component: search ingredient', () => {
  let component;
  let vm;
  let localVue = createLocalVue();

  localVue.use(Vuex);

  const state = {
    recipeIngrdients: [],
  };

  const recipeActions = {
    addIngredient: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();

    const recipeStore = {
      namespaced: true,
      state,
      actions: recipeActions,
    };

    const store = new Vuex.Store({
      modules: {
        recipe: recipeStore,
      },
    });

    component = mount(SearchIngredient, {
      stubs: {
        'v-text-field': '<input type="text">',
      },
      localVue,
      store,
    });

    vm = component.vm;
  });

  it('should add ingredients to recipe', (done) => {
    recipeActions.addIngredient.mockResolvedValue({});
    component.find('#search-ingredient-form').trigger('submit');
    component.setData({
      ingredient: 'apple',
      numberOfServing: 2,
    });

    expect(recipeActions.addIngredient).toHaveBeenCalled();
    vm.$nextTick(() => {
      expect(vm.isLoading).toBeFalsy();
      expect(vm.ingredient).toBe('');
      expect(vm.numberOfServing).toBe(1);
      done();
    });
  });

  it('should handle error', () => {
    recipeActions.addIngredient.mockRejectedValue(new Error({}));
    component.find('#search-ingredient-form').trigger('submit');

    vm.$nextTick().then(() => {
      expect(vm.isLoading).toBeFalsy();
    });
  });

  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});
