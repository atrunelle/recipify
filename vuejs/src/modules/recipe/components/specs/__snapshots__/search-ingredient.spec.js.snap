// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: search ingredient should match snapshot 1`] = `
<div class="search-ingredient v-card v-sheet theme--light">
  <div class="v-card__title v-card__title--primary">
    <h2 class="headline">What do you have in your fridge?</h2>
  </div>
  <form novalidate="novalidate" id="search-ingredient-form">
    <div class="layout justify-space-between align-center px-4 row"><input type="text" label="Ingredient and quantity" placeholder="Example: 200gr chicken" value="" class="ma-1"> <input type="text" label="Number of servings" value="1" class="ma-1"> <button disabled="disabled" type="submit" class="v-btn v-btn--disabled theme--light primary--text">
        <div class="v-btn__content">
          Add
        </div>
      </button></div>
  </form>
</div>
`;
